### PR TITLE
[Impeller] added a fallback that will make sure the blur fragment shader doesn't overflow

### PR DESCRIPTION
**Pull request by gaaclarke**
issue: https://github.com/flutter/flutter/issues/150462

This makes that usage never crash.  I would like to reevaluate for a better fix later though.  Something like:
1) Rework the downsample logic to make sure this gap doesn't exist.
1) Simply increase the kernel size.  We already did that before and technically we should be clamped to the 500 sigma so we won't have to do it forever.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by gaaclarke** [comment]
I'm not sure if this is going to pass CI.  The blur tests run fine, the added unit test is getting a stack overflow locally for some reason.

**Comment by gaaclarke** [comment]
Oh! we are doing a stack overflow since we temporarily will have more samples than will be supported, but then we use the lerp hack to make it fit.  I'm going to have to rework this, not minutes before my vacation.

**Comment by gaaclarke** [comment]
I got rid of the stack overflow.  I really would like more time to study this when I get back.  I'm going to refrain from landing it before I head out.  It's here if for some reason it becomes a priority.

File: impeller/entity/contents/filters/gaussian_blur_filter_contents.cc [hunk 1]
```diff
@@ -653,6 +650,18 @@ GaussianBlurPipeline::FragmentShader::KernelSamples GenerateBlurInfo(
     x_offset = 1;
   }
 
+  // This is a safe-guard to make sure we don't overflow the fragment shader.
+  // The kernel size is multiplied by 2 since we'll use the lerp hack on the
+  // result. In practice this isn't throwing away much data since the blur radii
+  // are around 53 before the down-sampling and max sigma of 500 kick in.
+  //
+  // TODO(https://github.com/flutter/flutter/issues/150462): Come up with a more
+  // wholistic remedy for this.  A proper downsample size should not make this
```
Comment (jonahwilliams) [inline impeller/entity/contents/filters/gaussian_blur_filter_contents.cc:659]: holistic ?

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by gaaclarke** [comment]
@jonahwilliams I tweaked this a bit since your approval if you want to give it another look.  Changes:
1) I created a new kernel samples struct that can hold 2x the samples (since the lerp hack will reduce it by 2).
1) I fixed an assert off-by-one error.
1) I added an assert in lerp hack.

File: impeller/entity/contents/filters/gaussian_blur_filter_contents.cc (hunk 1 above)
Comment (gaaclarke) [reply impeller/entity/contents/filters/gaussian_blur_filter_contents.cc:659]: Both spellings are accepted.  After a slight rabbit hole I can report that it isn't some sort of british-ism.  "Wholistic" came about about 100 years ago for some goofy reason, probably to sell snake oil.

**Pull request merged**